crates/orderbook/src/api/order_validation.rs

```diff
@@ -748,7 +784,7 @@ mod tests {
                     .await
                     .unwrap_err()
             ),
-            "InsufficientFunds"
+            "SellAmountOverflow"
```

Comment:
nit: completely unrelated to this PR (so feel free to not change that here), but this makes the test rely on how `#[derive(Debug)]` and seems a bit flaky to me... I think it would be much better to:
```rust
assert!(matches!(
    validator.validate_and_construct_order(...).await,
    Err(ValidationError::SellAmountOverflow),
));
```